quic/s2n-quic-transport/src/space/application.rs

```diff
@@ -522,6 +525,37 @@ impl<Config: endpoint::Config> ApplicationSpace<Config> {
 
         decrypted.map(|x| x.0)
     }
+
+    fn key_limits(max_mtu: MaxMtu) -> limited::Limits {
+        let mut limits = limited::Limits::default();
+
+        limits.max_mtu = max_mtu;
+
+        // AEAD optimizations are currently in the testing phase so make them opt-in at runtime
+        limits.sealer_optimization_threshold = {
+            static THRESHOLD: OnceCell<u64> = OnceCell::new();
+
+            *THRESHOLD.get_or_init(|| {
+                std::env::var("S2N_UNSTABLE_CRYPTO_OPT_TX")
+                    .ok()
+                    .and_then(|v| v.parse().ok())
+                    .unwrap_or(u64::MAX)
+            })
+        };
+
+        limits.opener_optimization_threshold = {
+            static THRESHOLD: OnceCell<u64> = OnceCell::new();
```

Comment:
When this is compiled, is the call to `OnceCell::new()` not part of the `key_limits` function at all?

Reply:
Yeah the static variable is actually initialized at compile-time (see [Static items](https://doc.rust-lang.org/reference/items/static-items.html)). This means there's a location allocated for a `OnceCell<u64>` in the application's data section. The `get_or_init` checks this location to determine if it has been initialized. If it hasn't, it executes the closure.